deltacat/utils/ray_utils/concurrency.py

```diff
@@ -107,4 +107,11 @@ def round_robin_options_provider(
     assert resource_keys, f"No resource keys given to round robin!"
     resource_key_index = i % len(resource_keys)
     key = resource_keys[resource_key_index]
-    return {"resources": {key: resource_amount_provider(resource_key_index)}}
+    if kwargs.get("pg_config"):
```

Comment:
we also need to update bundle index within the pg_config right?

Reply:
yeah, that will be next pr. This pr only tries to refactor the execute_compaction_round and make it working with existing pg which assumes node_id as resource key.